org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/util/gradle/PublishedGradleVersionsWrapper.java

```diff
@@ -18,58 +18,60 @@
 import org.eclipse.core.runtime.IStatus;
 import org.eclipse.core.runtime.Status;
 import org.eclipse.core.runtime.jobs.Job;
-
+import org.osgi.service.component.annotations.Component;
 import org.eclipse.buildship.core.internal.CorePlugin;
 import org.eclipse.buildship.core.internal.util.gradle.PublishedGradleVersions.LookupStrategy;
 
 /**
- * Wraps the {@link PublishedGradleVersions} functionality in a background job that handles all
- * exceptions gracefully. If an exception occurs while calling the underlying
- * {@link PublishedGradleVersions} instance, default version information is provided. This handles,
- * for example, those scenarios where the versions cannot be retrieved because the user is behind a
- * proxy or offline.
+ * Wraps the {@link PublishedGradleVersions} functionality in a background job
+ * that handles all exceptions gracefully. If an exception occurs while calling
+ * the underlying {@link PublishedGradleVersions} instance, default version
+ * information is provided. This handles, for example, those scenarios where the
+ * versions cannot be retrieved because the user is behind a proxy or offline.
  */
+@Component(property = { "service.ranking:Integer=1" }, service = PublishedGradleVersionsWrapper.class)
```

Comment:
Do you really need a non default service rank here?

Reply:
> Do you really need a non default service rank here?

If you look at the activator it also sets this ranking so I just replicated it.

Reply:
This is a final class, it is very unlikley that there are more than one service and even if they seem to be totaly eqivialent so this is most probably obsolete.

Service ranking is only important if there are multiple service of the same type and you need to have some kind of ordering of them.

Reply:
> This is a final class, it is very unlikley that there are more than one service and even if they seem to be totaly eqivialent so this is most probably obsolete.
> 
> Service ranking is only important if there are multiple service of the same type and you need to have some kind of ordering of them.

Agree, but changing logic AND implementation is IMHO a bad way. So leaving the priority in should allow @donat to review this easier. And next step could be to remove with a follow-up commit.